Deployment is failing due to deprecation of nexus-publish-plugin

**Description**
The [OSSRH workflow](https://github.com/meilisearch/meilisearch-java/actions/runs/6175913305/job/16782421217) failed during the v0.11.2 deployment.
This is caused by the deprecation of the [nexus-publish-plugin](https://github.com/marcphilipp/nexus-publish-plugin/) which is used by [gradle-nexus-staging-plugin](https://github.com/Codearte/gradle-nexus-staging-plugin) that we currently use in the `build.gradle` file.
The nexus-publish-plugin is now replaced by [this publish plugin](https://github.com/gradle-nexus/publish-plugin).

**Screenshots or Logs**
```
> Task :closeRepository
DEPRECATION WARNING. The staging repository ID is not provided. The fallback mode may impact release reliability and is deprecated. Please consult the project FAQ how it can be fixed.
Attempt 61/61 failed. WrongNumberOfRepositories was thrown with message 'Wrong number of received repositories in state 'open'. Expected 1, received 3'. Giving up. Configure longer timeout if necessary.

> Task :closeRepository FAILED

FAILURE: Build failed with an exception.

* What went wrong:
1 actionable task: 1 executed
Execution failed for task ':closeRepository'.
> Wrong number of received repositories in state 'open'. Expected 1, received 3

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org/

BUILD FAILED in 5m 11s
Error: Process completed with exit code 1.
```

**How to fix it**

You should be able to follow this guide for the [migration](https://github.com/gradle-nexus/publish-plugin/wiki/Migration-from-gradle_nexus_staging-plugin---nexus_publish-plugin-duo) and modify the `build.gradle` file with the new plugin

--- a/.github/workflows/ossrh-publish.yml
+++ b/.github/workflows/ossrh-publish.yml
@@ -34,12 +34,12 @@ jobs:
           echo "${{secrets.SIGNING_SECRET_KEY_RING_FILE}}" > ~/.gradle/secring.gpg.b64
           base64 -d ~/.gradle/secring.gpg.b64 > ~/.gradle/secring.gpg
       - name: Publish
-        run: ./gradlew publish -Psigning.keyId=${{secrets.SIGNING_KEY_ID}} -Psigning.password=${{secrets.SIGNING_PASSWORD}} -Psigning.secretKeyRingFile=$(echo ~/.gradle/secring.gpg)
+        run: ./gradlew publishToSonatype -Psigning.keyId=${{secrets.SIGNING_KEY_ID}} -Psigning.password=${{secrets.SIGNING_PASSWORD}} -Psigning.secretKeyRingFile=$(echo ~/.gradle/secring.gpg)
         env:
           OSSRH_USERNAME: ${{secrets.OSSRH_USERNAME}}
           OSSRH_PASSWORD: ${{secrets.OSSRH_PASSWORD}}
       - name: Close and release Nexus Repository
-        run: ./gradlew closeAndReleaseRepository
+        run: ./gradlew closeAndReleaseStagingRepository
         env:
           OSSRH_USERNAME: ${{secrets.OSSRH_USERNAME}}
           OSSRH_PASSWORD: ${{secrets.OSSRH_PASSWORD}}

--- a/.github/workflows/ossrh-publish.yml
+++ b/.github/workflows/ossrh-publish.yml
@@ -33,13 +33,9 @@ jobs:
         run: |
           echo "${{secrets.SIGNING_SECRET_KEY_RING_FILE}}" > ~/.gradle/secring.gpg.b64
           base64 -d ~/.gradle/secring.gpg.b64 > ~/.gradle/secring.gpg
-      - name: Publish
-        run: ./gradlew publishToSonatype -Psigning.keyId=${{secrets.SIGNING_KEY_ID}} -Psigning.password=${{secrets.SIGNING_PASSWORD}} -Psigning.secretKeyRingFile=$(echo ~/.gradle/secring.gpg)
+      - name: Publish, close and release Nexus Repository
+        run: ./gradlew publishToSonatype closeAndReleaseStagingRepository -Psigning.keyId=${{secrets.SIGNING_KEY_ID}} -Psigning.password=${{secrets.SIGNING_PASSWORD}} -Psigning.secretKeyRingFile=$(echo ~/.gradle/secring.gpg)
         env:
           OSSRH_USERNAME: ${{secrets.OSSRH_USERNAME}}
           OSSRH_PASSWORD: ${{secrets.OSSRH_PASSWORD}}
-      - name: Close and release Nexus Repository
-        run: ./gradlew closeAndReleaseStagingRepository
-        env:
-          OSSRH_USERNAME: ${{secrets.OSSRH_USERNAME}}
-          OSSRH_PASSWORD: ${{secrets.OSSRH_PASSWORD}}
+          SONATYPE_STAGING_PROFILE_ID: ${{secrets.SONATYPE_STAGING_PROFILE_ID}}

--- a/build.gradle
+++ b/build.gradle
@@ -12,7 +12,7 @@ plugins {
 	id 'maven-publish'
 	id 'signing'
 	id 'jacoco'
-	id 'io.codearte.nexus-staging' version '0.30.0'
+	id 'io.github.gradle-nexus.publish-plugin' version "1.3.0"
 	id "com.diffplug.spotless" version "6.13.0"
 }
 
@@ -170,6 +170,19 @@ publishing {
 	}
 }
 
+nexusPublishing {
+	packageGroup = "com.meilisearch"
+
+	repositories {
+		sonatype {
+			nexusUrl.set(uri("https://oss.sonatype.org/service/local/"))
+
+			username = System.getenv('OSSRH_USERNAME')
+			password = System.getenv('OSSRH_PASSWORD')
+		}
+	}
+}
+
 signing {
 	sign publishing.publications.mavenJava
 }
@@ -180,12 +193,6 @@ javadoc {
 	}
 }
 
-nexusStaging {
-	packageGroup = "com.meilisearch"
-	serverUrl = "https://oss.sonatype.org/service/local/"
-	username System.getenv('OSSRH_USERNAME')
-	password System.getenv('OSSRH_PASSWORD')
-}
 spotless {
 
 	java {

--- a/build.gradle
+++ b/build.gradle
@@ -179,6 +179,7 @@ nexusPublishing {
 
 			username = System.getenv('OSSRH_USERNAME')
 			password = System.getenv('OSSRH_PASSWORD')
+			stagingProfileId = System.getenv('SONATYPE_STAGING_PROFILE_ID')
 		}
 	}
 }

